### ISSUE TITLE
deepDuplicateArray

### ISSUE THREAD

**Issue by CMEONE** [TogaTech/helpful.js#28]
We should have a method that duplicates an array and all subarrays, like this:
```javascript
helpful.deepDuplicateArray([1, 2, [3, 4], 5]) => [1, 2, [3, 4], 5]
```
We encourage using the existing `duplicateArray` method and recursion to create the `deepDuplicateArray` method.

**Comment by TheSoniix** [comment]
Hi, I'm pretty new to github and would like to try my hand at this issue. During my computer science studies I worked a lot with JavaScript / Typescript and I think I can fix this issue. I would be happy if I would be assigned to that.

Best regards
Soniix

**Comment by CMEONE** [comment]
@TheSoniix Sounds great, we look forward to seeing your PR!

**Comment by TheSoniix** [comment]
How should I deal with objects in an array? Can i ignore that case, because we want a solution for subarrays?  We have not implemented a function of duplicating objects yet. The duplicateArray function would only saves the references of the objects.

**Comment by CMEONE** [comment]
Good point! #29 proposes duplication and deep duplication for an object. If you would like to work on that as well, please let me know so that I can assign you to that PR. I think both methods will need to use each other (for example, arrays within an object would use the `deepDuplicateArray` method and then objects within an array would need the `deepDuplicateObject` method) to ensure full duplication. However, custom class objects should not be duplicated and should just be transferred over as normal in both cases. You can detect if an object is not made by a custom class by checking if the constructor is an object constructor (`({}).constructor == myObject.constructor`).

**Comment by TheSoniix** [comment]
Of course! I will try it

**Comment by dwp23** [comment]
Hey! 🙋‍♀️
I'm currently learning JavaScript and would love to work on this issue as a first contribution.

Just to confirm — the goal is to implement a method like helpful.deepDuplicateArray([1, 2, [3, 4], 5]) that returns a deep copy of the array and all its subarrays, right?

Looking forward to your feedback! 😊
